segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageTests.java

```diff
@@ -3450,4 +3471,82 @@ public void close() throws Exception {
             this.metadataStore = null;
         }
     }
+
+
+    /**
+     * Runs all {@link ChunkedSegmentStorageTests} with all self checks on. It uses {@InMemoryChunkStorage}.
+     */
+    public static class ChunkedSegmentStorageTestsWithSelfCheck extends ChunkedSegmentStorageTests {
```

Comment:
@RaulGracia Here we are running full ChunkedSegmentStorage unit tests with data integrity checks on.
It uses InMemoryChunkStorage

Reply:
NoOp chunk storage + data integrity checks are not treated as an invalid combination, instead we simply skip validation in this case.

Note that `NoOpChunkStoragee` is used in unit tests instead of `InMemoryChunkStorage` to test very large chunks without allocating any memory.